src/pages/nerd-days.js

```diff
@@ -1,47 +1,15 @@
-import React, { useEffect } from 'react';
+import React from 'react';
 import { Link } from 'gatsby';
 import cx from 'classnames';
 import PageLayout from '../components/PageLayout';
+import MarketoForm from '../components/MarketoForm';
+import useMarketoForm from '../hooks/useMarketForm';
 import SEO from '../components/Seo';
 import nerdDays from '../images/nerd-days/nerd-days.png';
 import styles from './nerd-days.module.scss';
-import '../components/marketo.scss';
 
 const NerdDaysPage = () => {
-  useEffect(() => {
-    window.MktoForms2.loadForm('//app-abj.marketo.com', '412-MZS-894', 3525);
-
-    const pollForDefinition = function (scope, varname, callback) {
-      if (typeof scope[varname] !== 'undefined') {
-        return callback();
-      }
-      const interval = setInterval(function () {
-        if (typeof scope[varname] !== 'undefined') {
-          clearInterval(interval);
-          callback();
-        }
-      }, 250);
-    };
-    const script = document.createElement('script');
-    script.src = 'https://marketo.clearbit.com/assets/v1/marketo/forms.js';
-    script.async = true;
-    script.setAttribute(
-      'data-clearbit-publishable-key',
-      'pk_4d10daa544de6f993a9a9ce002ccd1c6'
-    );
-    script.onerror = function () {
-      // eslint-disable-next-line no-console
-      console.log('Clearbit Form JS unable to load');
-      pollForDefinition(window, 'MktoForms2', function () {
-        window.MktoForms2.whenReady(function (form) {
-          form.setValues({
-            clearbitFormStatus: 'Clearbit Form JS unable to load',
-          });
-        });
-      });
-    };
-    document.body.append(script);
-  }, []);
+  useMarketoForm('412-MZS-894', 3525, 'pk_4d10daa544de6f993a9a9ce002ccd1c6');
```

Comment:
Looking at this, I think it might make sense to move this hook call inside the `MarketoForm` component and pull these values from its props. This would allow us to use a single component for future events without needing to remember to use both the hook and the component.

```js
const MarketoForm = ({ munchkinId, id, publishableKey }) => {
  useMarketoForm(muchkinId, id, publishableKey)

  return // ...
}

// usage
<MarketoForm 
  id={3525} 
  munchkinId="412-MZS-894" 
  publishableKey="pk_4d10daa544de6f993a9a9ce002ccd1c6" 
/>
```

One nice side effect of that is that you don't have to repeat the `id` between the hook and the component, which might be error prone.

Thoughts?

Reply:
Great suggestion! Being able to use a single component would be best for lowering the number of errors in the future.